proto/frontend/src/device_mgr.cpp

```diff
@@ -475,10 +475,17 @@ class DeviceMgrImp {
   Status direct_meter_write(p4::Update_Type update,
                             const p4::DirectMeterEntry &meter_entry,
                             const SessionTemp &session) {
-    if (!check_p4_id(meter_entry.meter_id(), P4ResourceType::METER))
-      return make_invalid_p4_id_status();
-
+    if (!meter_entry.has_table_entry()) {
+      RETURN_ERROR_STATUS(Code::UNIMPLEMENTED,
+                          "Modifying ALL direct meters is not supported yet");
+    }
     const auto &table_entry = meter_entry.table_entry();
+    if (table_entry.table_id() == 0) {
```

Comment:
I don't think we should have 2 ways of "modifying all direct meters": missing `table_entry` (default) and `table_entry` with a default `table_id` of 0. Maybe we should agree that `table_entry` is always required?

Reply:
Fine with me. Fixed.